### PR TITLE
Hides expiry date for ads

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves https://github.com/brave/brave-browser/issues/4042

UI implementation: https://github.com/brave/brave-ui/pull/444

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests && npm run test-security`) on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Test Plan:
plan 1:
- clean profile
- enable rewards
- claim grant from settings page and panel and make sure that it shows expiration date
- make sure that you also see expiration date in the wallet under details (settings and panel)

plan 2:
- clean profile
- enable rewards
- view some ads
- get ads grant
- make sure that you don't see expiration date on the final step
- make sure that you don't see expiration date in the wallet under details (settings and panel)

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

File: components/brave_rewards/resources/ui/components/grant.tsx [hunk 1]
```diff
@@ -156,6 +157,10 @@ class Grant extends React.Component<Props, State> {
       tokens = convertProbiToFixed(grant.probi)
     }
 
+    if (grant.type === 'ads') {
```
Comment (tmancey) [inline components/brave_rewards/resources/ui/components/grant.tsx]: nit: logic is opposite to other places in the codebase where we would have the following code which removes the need to calculate the date for ads:

```
let date = ''
if (grant.type !== 'ads') {
    date = new Date(grant.expiryTime).toLocaleDateString()
}
```

File: components/brave_rewards/resources/ui/components/grant.tsx [hunk 2]
```diff
@@ -145,6 +145,7 @@ class Grant extends React.Component<Props, State> {
     let type
     let promoId
     let tokens = '0.0'
+    let date = new Date(grant.expiryTime).toLocaleDateString()
```
Comment (tmancey) [inline components/brave_rewards/resources/ui/components/grant.tsx]: Other places in this file `grant.expiryTime` is multiplied by `1000`???

File: components/brave_rewards/resources/ui/components/pageWallet.tsx [hunk 3]
```diff
@@ -156,9 +156,14 @@ class PageWallet extends React.Component<Props, State> {
     }
 
     return grants.map((grant: Rewards.Grant) => {
+      let expireDate = ''
```
Comment (tmancey) [inline components/brave_rewards/resources/ui/components/pageWallet.tsx]: nit: same logic is used in a couple places, consider decoupling into a method?

File: components/brave_rewards/resources/ui/components/grant.tsx (hunk 2 above)
Comment (NejcZdovc) [reply components/brave_rewards/resources/ui/components/grant.tsx]: for single grant we already multiply them in the reducer, but for grant list we don't do this

File: components/brave_rewards/resources/ui/components/grant.tsx (hunk 1 above)
Comment (NejcZdovc) [reply components/brave_rewards/resources/ui/components/grant.tsx]: done

File: components/brave_rewards/resources/ui/components/pageWallet.tsx (hunk 3 above)
Comment (NejcZdovc) [reply components/brave_rewards/resources/ui/components/pageWallet.tsx]: we only use it here for the settings page, sadly extension code has different util files

**Pull request merged**